core/templates/domain/blog-admin/blog-admin-backend-api.service.ts

```diff
@@ -0,0 +1,130 @@
+}
+
+interface RoleToActionsBackendResponse {
+  [role: string]: string[];
+}
+
+interface ConfigPropertiesBackendResponse {
+  [property: string]: Object;
+}
+
+interface ConfigPropertyValues {
+  'max_number_of_tags_assigned_to_blog_post': number,
+  'list_of_default_tags_for_blog_post': string[]
+}
+
+export interface BlogAdminPageDataBackendDict {
+  'config_properties': ConfigPropertiesBackendResponse;
+  'role_to_actions': RoleToActionsBackendResponse;
+  'updatable_roles': UserRolesBackendResponse;
+}
+
+export interface BlogAdminPageData {
+  configProperties: ConfigPropertiesBackendResponse;
+  roleToActions: RoleToActionsBackendResponse;
+  updatableRoles: UserRolesBackendResponse;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class BlogAdminBackendApiService {
+  constructor(private http: HttpClient) {}
+
+  async getDataAsync(): Promise<BlogAdminPageData> {
+    return new Promise((resolve, reject) => {
+      this.http.get<BlogAdminPageDataBackendDict>(
+        '/blogadminhandler').toPromise().then(response => {
+        resolve({
+          updatableRoles: response.updatable_roles,
+          roleToActions: response.role_to_actions,
+          configProperties: response.config_properties,
+        });
+      }, errorResponse => {
+        reject(errorResponse.error.error);
+      });
+    });
+  }
+
+  private async _postRequestAsync(
+      handlerUrl: string, payload: Object, action?: string): Promise<void> {
+    return new Promise((resolve, reject) => {
+      this.http.post<void>(
+        handlerUrl, { action, ...payload }).toPromise()
+        .then(response => {
+          resolve(response);
+        }, errorResponse => {
+          reject(errorResponse.error.error);
+        });
+    });
+  }
+
+  async saveConfigPropertiesAsync(
+      newConfigPropertyValues: ConfigPropertyValues): Promise<void> {
+    let action = 'save_config_properties';
+    let payload = {
+      new_config_property_values: newConfigPropertyValues
+    };
+    return this._postRequestAsync('/blogadminhandler', payload, action);
+  }
+
+  async revertConfigPropertyAsync(configPropertyId: string): Promise<void> {
+    let action = 'revert_config_property';
+    let payload = {
+      config_property_id: configPropertyId
+    };
+    return this._postRequestAsync('/blogadminhandler', payload, action);
+  }
+
+  async updateUserRoleAsync(
+      newRole: string, username: string): Promise<void> {
+    let payload = {
+      role: newRole,
+      username: username,
+    };
+    return this._postRequestAsync('/blogadminrolehandler', payload);
+  }
+
+  async removeBlogEditorAsync(username: string): Promise<void> {
+    return new Promise((resolve, reject) => {
+      this.http.put<void>(
+        '/blogadminrolehandler', {
+          username: username,
+        }
+      ).toPromise().then(response => {
+        resolve(response);
+      }, errorResponse => {
+        reject(errorResponse.error.error);
+      });
+    });
+  }
```

Comment:
This request doesn't have a response. Maybe you can remove the extra wrapping with new Promise.
```suggestion
  async removeBlogEditorAsync(username: string): Promise<void> {
    this.http.put<void>(
      '/blogadminrolehandler', {
        username: username,
      }
    ).toPromise()
  }
```
If returning the errorCode is important, I would strongly recommend using Observables and not converting it to Promise.
Here is how it would go:
 ```suggestion
  async removeBlogEditorAsync(username: string): Observable<void> {
    this.http.put<void>(
      '/blogadminrolehandler', {
        username: username,
      }
    ).pipe(catchError(err => throwError(err.error.error));
  }
```
Note that pipe, catchError and throwError are from the RxJS library.

Reply:
Done.